### ISSUE TITLE
Different NameFormat Types 

### ISSUE THREAD

**Issue by bennom** [brigade/scss-lint#879]
Hello! First of all: thank you for this great tool. I don't want to miss it in some of my projects :-)

I came across the issue, that I can set `NameFormat` convention to (eg) `hyphenated_lowercase`, but when I also want to set `function_convention` and `mixin_convetion` to `camel_case`, the linter still complains that the function and mixin should be written in lowercase with hyphens.

My yml:
```yml
NameFormat:
    enabled: true
    variable_convention: hyphenated_lowercase
    function_convention: camel_case
    mixin_convetion: camel_case
```

Any ideas why this is happening? Thanks for your help in advance! :-)

PS: I am using the current `scss-lint` version 0.51.0 (and Sass 3.4.22 on macOS Sierra 10.12.1).

**Comment by trotzig** [comment]
This sounds like a bug to me. I don't have time to look into this myself, but we're definitely open for a PR fixing this.

**Comment by sds** [comment]
Hey @bennom, I can't reproduce this behavior with the latest version of SCSS-Lint (0.54.0 at time of writing).

Make sure you have the `NameFormat` linter under the `linters` key, i.e.

```yaml
linters:
  NameFormat:
    enabled: true
    ...
```

Furthermore, you misspelled `mixin_convetion` (missing an `n`).

Using the following `.scss-lint.yml`:

```yaml
linters:
  NameFormat:
    enabled: true
    variable_convention: hyphenated_lowercase
    function_convention: camel_case
    mixin_convention: camel_case
```

...against the following SCSS file:

```scss
$myVar: 'hello';

@function my-func() {
}

@mixin my-mixin() {
}
```

...I get the following output:

```
test.scss:1:2 [W] NameFormat: Name of variable `myVar` should be written in all lowercase letters with hyphens instead of underscores
test.scss:3:1 [W] NameFormat: Name of function `my-func` should be written in camelCase format
test.scss:6:1 [W] NameFormat: Name of mixin `my-mixin` should be written in camelCase format
```

Hope that helps!
